appinventor/components/src/com/google/appinventor/components/runtime/util/JavaStringUtils.java

```diff
@@ -290,18 +310,28 @@ public static String replaceAllMappings(String text, Map<Object, Object> mapping
    */
   private static String applyMappings(String text, Map<String, String> mappings, List<String> keys) {
     // Create a set of ranges to keep track of which index ranges in the
-    // original text string are already set for replacement.
-    TreeRangeSet<Integer> ranges = TreeRangeSet.create();
-
-    // Map to map Range to String to replace with.
-    // E.g. [1, 3) -> 'abcd' indicates that the substring from 1 to 3 exclusive should
-    // be replaced with the string 'ab'.
-    // Note that the length of the string does not matter.
-    Map<Range<Integer>, String> replacements = new TreeMap<>(new Comparator<Range<Integer>>() {
+    // original text string are already set for replacement, together with the
+    // text to replace with. We sort this TreeSet in descending order to preserve
+    // indices of all ranges after replacement.
+    TreeSet<Range> ranges = new TreeSet<Range>(new Comparator<Range>() {
```

Comment:
Minor nit: You could make this `static final` on the class so we only instantiate one.

Reply:
You mean the comparator, right?

Reply:
Yes, the comparator only needs to be constructed once.

Reply:
Done!